addons/account/tests/test_account_move_out_invoice.py

```diff
@@ -3692,3 +3692,20 @@ def test_post_valid_invoices_when_auto_post(self):
         self.assertEqual(invalid_invoice_2.state, 'draft')
         self.assertEqual(invalid_invoice_2.message_ids[0].body,
                          "<p>The move could not be posted for the following reason: You need to add a line before posting.</p>")
+
+    def test_taxes_on_payment_term_line(self):
+        ''' Test that no tax is populated on the "payment_term" line of an invoice.
+        '''
+
+        receivable_account = self.partner_a.property_account_receivable_id
+        receivable_account.write({'tax_ids': [(6, 0, self.company_data['default_tax_sale'].ids)]})
+
+        move_form = Form(self.env['account.move'].with_context(default_move_type='out_invoice'))
+        move_form.partner_id = self.partner_a
+        with move_form.invoice_line_ids.new() as line_form:
+            line_form.product_id = self.product_a
+        move = move_form.save()
+
+        payment_term_line = move.line_ids.filtered(lambda l: l.display_type == 'payment_term')
+        self.assertEqual(payment_term_line.account_id.id, receivable_account.id)
+        self.assertEqual(len(payment_term_line.tax_ids), 0)
```

Comment:
Not mandatory but we could use `assertRecordValues` to ass both at the same time